cmd/livegrep-fetch-reindex/main.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"sync"
+)
+
+type IndexConfig struct {
+	Name         string       `json:"name"`
+	Repositories []RepoConfig `json:"repositories"`
+}
+
+type RepoConfig struct {
+	Path      string            `json:"path"`
+	Name      string            `json:"name"`
+	Revisions []string          `json:"revisions"`
+	Metadata  map[string]string `json:"metadata"`
+}
+
+var (
+	flagIndexPath   = flag.String("out", "livegrep.idx", "Path to write the index")
+	flagRevparse    = flag.Bool("revparse", true, "whether to `git rev-parse` the provided revision in generated links")
+	flagSkipMissing = flag.Bool("skip-missing", false, "skip repositories where the specified revision is missing")
+)
+
+const Workers = 8
+
+func main() {
+	flag.Parse()
+	log.SetFlags(0)
+
+	if len(flag.Args()) != 1 {
+		log.Fatal("Expected exactly one argument (the index json configuration)")
+	}
+
+	data, err := ioutil.ReadFile(flag.Arg(0))
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+
+	var cfg IndexConfig;
+	if err = json.Unmarshal(data, &cfg); err != nil {
+		log.Fatalf("reading %s: %s", flag.Arg(0), err.Error())
+	}
+
+	err = exec.Command("ssh-add", "-l").Run()
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	if err := checkoutRepos(&cfg.Repositories); err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	tmp := *flagIndexPath + ".tmp"
+
+	args := []string{
+		"--debug=ui",
+		"--dump_index",
+		tmp,
+		"--index_only",
+	}
+	if *flagRevparse {
+		args = append(args, "--revparse")
+	}
+	args = append(args, flag.Arg(0))
+
+	cmd := exec.Command(
+		path.Join(path.Dir(path.Dir(path.Dir(os.Args[0]))), "src", "tools", "codesearch"),
```

Comment:
Yeah, I haven't been using `bazel run` because then (e.g.) the sandbox prevents me from accessing external git repositories, etc. And so I haven't bothered to use the runfiles directory at all.

I think I'll default both of them to 'same directory' but add a `-codesearch` flag to give an out for debugging or alternate deployments.